README.md

```diff
@@ -0,0 +1,57 @@
+TinyVPN
+=======
+Code named petrel.
+
+Architecture
+------------
+### Logic Layers
+1. Authentication Service: 
+    1. Exchange username and password for session key
+    2. Put session key into client-session map
+2. Transport Layer Listener: 
+    1. TCP or UDP socket listener, accepts connection/packets from the client
+    2. Extract session key to create / load corresponding client, maintains client-session map
+    3. Put packets into a packets channel
+    4. Accepts packets from packets return channel, lookup the corresponding client connection and return the packets
```

Comment:
In layer 2, should we call the channel "packets channel" or "packets return channel"? I think there are two things to consider:
a. There is a difference in encrypted channel an unencrypted channel.
b. The naming of "packets channel" and "packets return" channel does not say a lot about themselves.
I suggest changing the name to something like "pubChanOut" "pubChanIn" and "privChanIn", "privChanOut" with encrypted indication.